unitTests/HostMatrix.C

```diff
@@ -0,0 +1,212 @@
+// SPDX-FileCopyrightText: 2024 OGL authors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+
+#include "OGL/MatrixWrapper/HostMatrix.H"
+
+#include "gtest/gtest.h"
+
+//---------------------------------------------
+// some_header.h
+extern int my_argc;
+extern char **my_argv;
+// eof
+//---------------------------------------------
+
+//---------------------------------------------
+// main.cpp
+int my_argc;
+char **my_argv;
+
+
+class HostMatrixEnvironment : public testing::Environment {
+public:
+    void SetUp()
+    {
+        dict.add("executor", "reference");
+
+        args_ = std::make_shared<Foam::argList>(my_argc, my_argv);
+        runTime_ = std::make_shared<Foam::Time>("controlDict", *args_.get());
+
+        mesh = std::make_shared<Foam::fvMesh>(
+            Foam::IOobject(word{""}, runTime_->timeName(), *runTime_.get(),
+                           Foam::IOobject::MUST_READ),
+            false);
+
+        exec = std::make_shared<ExecutorHandler>(runTime_->thisDb(), dict,
+                                                 "dummy", true);
+
+        // delete listener on ranks != 0
+        // to clean up output
+        ::testing::TestEventListeners &listeners =
+            ::testing::UnitTest::GetInstance()->listeners();
+        if (Foam::Pstream::myProcNo() != 0) {
+            delete listeners.Release(listeners.default_result_printer());
+        }
+
+        word fieldName{"p"};
+        dimensionSet ds{};
+        field = std::make_shared<
+            GeometricField<scalar, Foam::fvPatchField, Foam::volMesh>>(
+            Foam::IOobject(fieldName, runTime_->timeName(), runTime_->thisDb(),
+                           Foam::IOobject::MUST_READ),
+            *mesh.get(), ds);
+
+        Foam::fvMatrix<scalar> fvMatrix{*field.get(), ds};
+
+        interfaces = field->boundaryField().scalarInterfaces();
+
+        hostMatrix = std::make_shared<HostMatrixWrapper>(
+            *exec.get(), runTime_->thisDb(), mesh->lduAddr(), true,
+            fvMatrix.diag().data(), fvMatrix.upper().data(),
+            fvMatrix.lower().data(), fvMatrix.boundaryCoeffs(),
+            fvMatrix.internalCoeffs(), interfaces, dict, "fieldName", 0);
+    }
+
+    Foam::lduInterfaceFieldPtrsList interfaces;
+    Foam::PtrDynList<Foam::lduInterfaceField> newInterfaces;
+    std::shared_ptr<Foam::argList> args_;
+    std::shared_ptr<Foam::Time> runTime_;
+    std::shared_ptr<fvMesh> mesh;
+    Foam::dictionary dict;
+    std::shared_ptr<GeometricField<scalar, Foam::fvPatchField, Foam::volMesh>>
+        field;
+    std::shared_ptr<const ExecutorHandler> exec;
+    std::shared_ptr<const HostMatrixWrapper> hostMatrix;
+};
+
+const testing::Environment *global_env =
+    AddGlobalTestEnvironment(new HostMatrixEnvironment);
+
+TEST(HostMatrix, hasCorrectNumberArgsSize)
+{
+    EXPECT_EQ(((HostMatrixEnvironment *)global_env)->args_->size(), 1);
+}
+
+TEST(HostMatrix, runsInParallelWithCorrectNumberOfRanks)
```

Comment:
Since the unit tests for host matrix wrapper are hardcoded at the moment, it's better to put this test to a place where setup takes place, e.g., the body of the `SetUp` function

Reply:
Aggree, will do that.